api/v1beta1/azuremanagedcontrolplanetemplate_webhook.go

```diff
@@ -88,12 +88,6 @@ func (mcpw *azureManagedControlPlaneTemplateWebhook) ValidateUpdate(ctx context.
 	if !ok {
 		return nil, apierrors.NewBadRequest("expected an AzureManagedControlPlaneTemplate")
 	}
-	if err := webhookutils.ValidateImmutable(
```

Comment:
That's what I suspect would happen.

Reply:
Actually, from my experience, CAPZ will fail to reconcile an AzureManagedControlPlane with an empty subscriptionID (although the field is _optional_ in the API!).

```
capz-controller-manager-667d8ff489-drt8x E0208 00:25:43.520687       1 controller.go:329]  "msg"="Reconciler error" "error"="error creating AzureManagedControlPlane default/dlaksdkp27: failed to reconcile AzureManagedControlPlane service managedcluster: failed to create resource dlaksdkp27/dlaksdkp27 (service: managedcluster): Code=\"LinkedAuthorizationFailed\" Message=\"The client has permission to perform action 'Microsoft.Network/virtualNetworks/subnets/join/action' on scope '/subscriptions/REDACTED/resourceGroups/dlaksdkp27/providers/Microsoft.ContainerService/managedClusters/dlaksdkp27', however the linked subscription 'resourceGroups' was not found. \"" "AzureManagedControlPlane"={"name":"dlaksdkp27","namespace":"default"} "controller"="azuremanagedcontrolplane" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedControlPlane" "name"="dlaksdkp27" "namespace"="default" "reconcileID"="REDACTED"
``` 

I thought it would make sense to make subscriptionID mutable everywhere, but if we're concerned about behavior for AzureManagedControlPlane specifically, I think the field could remain immutable, because an empty (or invalid) value never worked to begin with.

Reply:
Since AzureManagedControlPlane can not be successfully reconciled  when subscriptionID is empty, I don't see a need to make it mutable in that resource. I'll update the PR to make it immutable.

Reply:
Done in e8a948c5cdc6da492f4c1bf296a257095646fb5e

Reply:
Done in e8a948c5cdc6da492f4c1bf296a257095646fb5e